Update joystick docs (throttle, expo, advanced)

Williangalvani opened the pull request:
Update joystick docs:
Describe throttle modes, expo, and advanced options.

the changes can be seen [here](https://github.com/Williangalvani/qgc-user-guide/blob/joystickOptions/en/SetupView/Joystick.md#throttle-options)

Williangalvani committed:
Setup: describe joystick throttle modes and expo

jaxxzer commented on en/SetupView/Joystick.md:
```diff
@@ -33,6 +33,41 @@ To configure a joystick:
+## Throttle Options
+   ![Joystick setup - Throttle Modes](../../assets/setup/joystick_throttle_modes.jpg)
+
+- **Center stick is zero throttle**: Centered or lowered stick sends 0, raised stick sends 1000.
```
```suggestion
- **Center stick is zero throttle**: Centered or lowered stick sends 0 in [MANUAL_CONTROL **z**](https://mavlink.io/en/messages/common.html#MANUAL_CONTROL), raised stick sends 1000.
```

jaxxzer replied:
same below

jaxxzer commented on en/SetupView/Joystick.md:
```diff
@@ -50,6 +50,25 @@ The expo slider allows you to make the sticks less sensitive in the center, allo
+    - **Force**: User controls the forces applied to the vehicle, mavlink [SET_POSITION_TARGET_LOCAL_NED](https://mavlink.io/en/messages/common.html#SET_POSITION_TARGET_LOCAL_NED) messages with bitmask for **force** only are used.
+    - **Velocity**: User controls the forces applied to the vehicle, mavlink [SET_POSITION_TARGET_LOCAL_NED](https://mavlink.io/en/messages/common.html#SET_POSITION_TARGET_LOCAL_NED) messages with bitmask for **velocity** only are used.
+
+- **Message Frequency**: This handles how often the joystick messages are sent. The default is 25 Hz, automatically lowered to 5Hz when there are no changes.
```
When the joystick is idle (inputs are not changing), the joystick commands are sent to the vehicle at 5Hz. When the joystick is in use (input values are changing), the joystick commands are sent to the vehicle at the (higher) frequency configured by this setting. The default is 25Hz.

jaxxzer commented on en/SetupView/Joystick.md:
```diff
@@ -50,6 +50,25 @@ The expo slider allows you to make the sticks less sensitive in the center, allo
+
+- **Message Frequency**: This handles how often the joystick messages are sent. The default is 25 Hz, automatically lowered to 5Hz when there are no changes.
+
+- **Enable Circle Correction**: Circle correction is used because joysticks usually don't have as much stick travel as RC radios, these can describe a square, while joysticks usually describe a circle. This options inscribes a square into the joystick movement area. This way it is possible to reach all four corners, but this costs some resolution as the effective stick travel is reduced.
```
Disabled: The joystick position is sent to the vehicle the way that it is read from the joystick device, unchanged. On some joysticks, the (roll, pitch) values are confined to the space of a circle inscribed inside of a square. In this figure, point B would command full pitch forward and full roll right, but the joystick is not able to reach point B because the retainer is circular.  This means that you will not be able to achieve full roll and pitch deflection simultaneously.
![](https://ds055uzetaobb.cloudfront.net/image_optimizer/04275a281b8176289a10fd007945dc59204bfeaa.jpg)
Enabled: The joystick values are adjusted in software to ensure full range of commands. The usable area of travel and resolution is decreased, however, because the area highlighted grey in the figure is no longer used.
![](https://ds055uzetaobb.cloudfront.net/image_optimizer/6ffee5b5f582f78d782b61cbdeca791657458164.jpg)

jaxxzer replied:
make a separate commit. maybe hamish likes less detail

jaxxzer commented on en/SetupView/Joystick.md:
```diff
@@ -50,6 +50,25 @@ The expo slider allows you to make the sticks less sensitive in the center, allo
+
+- **Enable Circle Correction**: Circle correction is used because joysticks usually don't have as much stick travel as RC radios, these can describe a square, while joysticks usually describe a circle. This options inscribes a square into the joystick movement area. This way it is possible to reach all four corners, but this costs some resolution as the effective stick travel is reduced.
+
+- **Deadbands**: Deadbands avoid that noise or small oscilations of the sticks are interpreted as commands. They can be adjusted in the first step of calibration, or by dragging vertically on the corresponding axis monitor.
```
Deadbands allow input changes to be ignored when the sticks are near their neutral positions. This helps to avoid noise or small oscilations on sensitive sticks are interpreted as commands, or small offsets when sticks do not re-center well. They can be adjusted during the first step of the [calibration](hyperlink), or by<s> dragging the vertical slider on the corresponding axis monitor.</s>

jaxxzer replied:
> dragging the vertical slider on the corresponding axis monitor.

I can't get this to work, don't know how it's supposed to, keep what you have (match the comment next to enable checkbox)

Williangalvani replied:
> I can't get this to work, don't know how it's supposed to, keep what you have (match the comment next to enable checkbox)

This is an issue with how tall the page is. It is tall enough that it enabled scrolling, so if you try to drag the mouse up or down it scrolls instead of changing the property.

I'll open an issue about that and maybe change it so it uses horizontal movement instead.

Williangalvani replied:
> or by dragging the vertical slider on the corresponding axis monitor.

There is actually no vertical slider, you were supposed to drag vertically the horizontal bar. It is weird.

Williangalvani replied:
I opened an issue https://github.com/mavlink/qgroundcontrol/issues/7214, I'll probably change the logic to drag horizontally instead, makes more sense to me. Or maybe the mouse wheel?

jaxxzer replied:
hmm ok I'll think more when I can try it again, ty

Williangalvani replied:
I kept "or by dragging vertically on the corresponding axis monitor." since it is what the inferface says, at least until that is changed to work by dragging horizontally.

Williangalvani committed:
Setup: improve Joystick circle correction documentation

jaxxzer commented on en/SetupView/Joystick.md:
```diff
@@ -33,6 +33,49 @@ To configure a joystick:
+
+    ![](../../assets/setup/joystick_circle_correction2.jpg)
+
+- **Deadbands**: Deadbands allow input changes to be ignored when the sticks are near their neutral positions. This helps to avoid noise or small oscilations on sensitive sticks are interpreted as commands, or small offsets when sticks do not re-center well. They can be adjusted during the first step of the [calibration](#configure), or by dragging vertically on the corresponding axis monitor.
```
```suggestion
- **Deadbands**: Deadbands allow input changes to be ignored when the sticks are near their neutral positions. This helps to avoid noise or small oscilations on sensitive sticks which may be interpreted as commands, or small offsets when sticks do not re-center well. They can be adjusted during the first step of the [calibration](#configure), or by dragging vertically on the corresponding axis monitor.
```

jaxxzer commented on en/SetupView/Joystick.md:
```diff
@@ -17,7 +17,7 @@ This is enabled by default for PX4 SITL builds (see the [Parameters](../SetupVie
 
 
-## Configuring the Joystick
+## Configuring the Joystick {#configure}
```
double check the render here (nothing in github md, suppose gitbook takes care of it)

Williangalvani replied:
I tested locally, it works.

jaxxzer committed:
Update joystick configuration instructions

1. hotplugging is supported, no need to shutdown qgc first
2. joystick should be selected before calibrating
3. settings should be confirmed before enabling output

Williangalvani committed:
Setup: add joystick advanced settings
